File: javascript/0070-climbing-stairs.js
/**
 * 70. Climbing Stairs
 * https://leetcode.com/problems/climbing-stairs/
 */

/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n) {
  if (n <= 3) return n;
  let prev = 2,
    last = 3;
  for (let i = 4; i <= n; i++) [prev, last] = [last, prev + last];
  return last;
};

File: swift/0070-climbing-stairs.swift
/**
 * 70. Climbing Stairs
 * https://leetcode.com/problems/climbing-stairs/
 **/

class Solution {
    func climbStairs(_ n: Int) -> Int {
        if n <= 3 {
            return n
        }

        var prev = 2, last = 3
        for _ in 4 ... n {
            (prev, last) = (last, last + prev)
        }

        return last
    }
}